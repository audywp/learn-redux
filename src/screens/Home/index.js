import {
  View,
  Text,
  Image,
  ActivityIndicator,
  TouchableOpacity,
  FlatList,
  StyleSheet,
} from 'react-native';
import React from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {ms} from 'react-native-size-matters';
import {IMAGE_URL} from '../../Helpers/api';
import {getDataMovie, getDetailMovieById} from './redux/action';

export default function Home() {
  const {loading, theme} = useSelector(state => state.Global);
  const style = StyleSheet.create({
    button: {color: theme === 'light' ? 'black' : 'white'},
  });
  const dispatch = useDispatch();

  const {nowPlaying = []} = useSelector(state => state.home);

  const getDataMovies = () => {
    dispatch(getDataMovie()); // dispatch untuk fetching
  };

  const getMovieByID = item => {
    dispatch(getDetailMovieById(item.id));
  };

  const MovieContainer = ({item}) => {
    return (
      <TouchableOpacity onPress={() => getMovieByID(item)}>
        <Image
          source={{uri: `${IMAGE_URL}${item.poster_path}`}}
          resizeMode="contain"
          style={{height: ms(200), width: ms(200)}}
        />
      </TouchableOpacity>
    );
  };

  if (loading) {
    return (
      <View flex={1} justifyContent="center" alignItems="center">
        <ActivityIndicator size={ms(120)} color="red" />
      </View>
    );
  }

  return (
    <View>
      <FlatList
        testID="list movie"
        data={nowPlaying}
        keyExtractor={item => item.id}
        renderItem={MovieContainer}
        ListEmptyComponent={() => {
          return <Text testID="empty state">There is no data</Text>;
        }}
      />
      <TouchableOpacity onPress={getDataMovies}>
        <Text style={style.button}>get data Now Playing</Text>
      </TouchableOpacity>
      <Text>Home</Text>
    </View>
  );
}
