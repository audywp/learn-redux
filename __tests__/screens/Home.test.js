import * as React from 'react';
import {create} from 'react-test-renderer';
import ContainerTesting from '../../src/Helpers/reduxTesting';
import Home from '../../src/screens/Home';

jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

describe('Name of the group', () => {
  describe('1', () => {
    test('should render', () => {
      const component = create(ContainerTesting(<Home />));
      const root = component.root;
      const {Global} = root.props.store.getState();
      expect(root.props.store).toBeTruthy();
      expect(Global).toBeTruthy();
      expect(typeof Global.token).toEqual('string');

      expect(component).toMatchSnapshot();
    });
  });
});
