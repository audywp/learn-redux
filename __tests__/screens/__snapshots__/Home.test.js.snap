// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group 1 should render 1`] = `null`;
